ERR_MODULE_NOT_FOUND with ESM imports

Hello, I'm really looking forward to testing this library but I ran into an import issue using TypeScript + esbuild + ESM.

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/user/code/project/node_modules/papr/esm/model' imported from /home/user/code/project/node_modules/papr/esm/index.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:416:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
  ```

Curious if anyone else has ran into this. I'm relatively new to using esbuild so could be some simple mistakes on my end. Builds with both tsc and esbuild succeed, this is happening when I run the build output.

My file for testing:

```typescript
import { MongoClient } from 'mongodb';
import Papr from 'papr';

const papr = new Papr();

const connection = await MongoClient.connect(
  'mongodb://localhost:27017',
);
papr.initialize(connection.db('tantor'));

await papr.updateSchemas();
```

--- a/build.sh
+++ b/build.sh
@@ -1,0 +1,15 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# ESM
+
+tsc --project ./tsconfig-esm.json
+
+sed -i "s/from '.\/\(.*\)';/from '.\/\1.js';/g" ./esm/*.js
+
+# CommonJS
+
+tsc --project ./tsconfig-cjs.json
+
+echo "{\"type\":\"commonjs\"}" > ./cjs/package.json

--- a/package.json
+++ b/package.json
@@ -38,7 +38,7 @@
     "url": "https://github.com/plexinc/papr/issues/new"
   },
   "scripts": {
-    "build": "tsc --project ./tsconfig-esm.json && tsc --project ./tsconfig-cjs.json && echo \"{\\\"type\\\":\\\"commonjs\\\"}\" > ./cjs/package.json",
+    "build": "./build.sh",
     "docs": "node docs/build.js && docsify serve ./docs",
     "benchmark": "yarn build && node --experimental-specifier-resolution=node ./benchmark/run.js",
     "lint": "eslint --config ./.eslintrc.ts.json --ext .ts ./example ./src && eslint --ext .js,.mjs ./benchmark ./tests",

--- a/tests/build.sh
+++ b/tests/build.sh
@@ -13,7 +13,7 @@ tar xf ./build/papr.tgz --directory=./build/
 cd tests/esm/
 rm -f package-lock.json
 npm install
-node --experimental-specifier-resolution=node ./index.js
+node ./index.js
 
 cd ../cjs/
 rm -f package-lock.json

